picasso/src/main/java/com/squareup/picasso/RequestCreator.java

```diff
@@ -373,8 +378,19 @@ public void fetch() {
 
       Request request = createRequest(started);
       String key = createKey(request, new StringBuilder());
+      Bitmap bitmap = picasso.quickMemoryCacheCheck(key);
```

Comment:
Its true. The same assumption applies even for other API calls that force main thread. They assume the cache implementation is thread safe.